### ISSUE TITLE
ssh-add did not remember the passphrase of the private key

### ISSUE THREAD

**Issue by chucklu** [git-for-windows/git#3499]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options
git version 2.33.1.windows.1
cpu: x86_64
built from commit: 05d80adb775077d673fab685009ede4a1003ed92
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.1288]

```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: FFOnly
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

I have configure git to use C:\Program Files\TortoiseGit\bin\TortoisePlink.exe along with C:\Program Files\TortoiseGit\bin\pageant.exe successfully. Now I have to switch to default openssh of git, as the Azure DevOps did not support multi ssh keys. 

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
https://github.com/git-for-windows/git/wiki/Auto-launching-ssh-agent-when-git-starts
use the manually section of the document to add the private key
```
 - What did you expect to occur after running these commands?

when I add the key successfully, git fetch will not ask for passphrase of the private key.

 - What actually happened instead?

git bash still ask for the passphrase of my private key

I am not sure how to revert the git for windows to initial state, any idea about this?

**Comment by dscho** [comment]
So, calling `ssh` manually also shows that prompt? If so, can you call `ssh` with `-v -v -v -v` and study the output to find something that looks like it might be the issue?

**Comment by chucklu** [comment]
What did you mean by  `calling ssh manually `?
I just configure the script mentioned in [Auto launching ssh agent when git starts](https://github.com/git-for-windows/git/wiki/Auto-launching-ssh-agent-when-git-starts)
```
# ssh-agent auto-launch (0 = agent running with key; 1 = w/o key; 2 = not run.)
agent_run_state=$(ssh-add -l >| /dev/null 2>&1; echo $?)
if   [ $agent_run_state = 2 ]; then
  eval $(ssh-agent -s)
  ssh-add ~/.ssh/id_rsa
elif [ $agent_run_state = 1 ]; then
  ssh-add ~/.ssh/id_rsa
fi
```
When I open git bash, it will popup and ask the passphrase of private key. And when i input the correct key, the private key was added to ssh agent, and can be confirmed by `ssh-add -l`.

Then I try to use `git fetch`, but it still ask the passphrase for the private key. I did not run ssh manually.
What ssh command I supposed to run for debug to get verbose detail?

**Comment by chucklu** [comment]
I got the ssh command by `GIT_TRACE=1 GCM_TRACE=1 git fetch origin`， then I execute the ssh command `ssh -o SendEnv=GIT_PROTOCOL git@github.com -v -v -v -v`

I resolve  the issue by delete the .ssh/conifg file, this file was added as [Azure DevOps required ](https://docs.microsoft.com/en-us/azure/devops/repos/git/use-ssh-keys-to-authenticate?view=azure-devops) to fix  `Q: I have multiple SSH keys. How do I use different SSH keys for different SSH servers or repos?`

wrong case
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 51


correct case
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 60

Now I configure the .bashrc like this, ssh-add two private key, and put the azure private key at the first order. The problem get resolved, have no idea how it works. @dscho  Thanks for you help.
```# ssh-agent auto-launch (0 = agent running with key; 1 = w/o key; 2 = not run.)
agent_run_state=$(ssh-add -l >| /dev/null 2>&1; echo $?)
if   [ $agent_run_state = 2 ]; then
  eval $(ssh-agent -s)
  ssh-add "personal folder\.ssh\001\id_rsa"
  ssh-add "personal folder\clu\.ssh\2021-1025\id_rsa"
elif [ $agent_run_state = 1 ]; then
  ssh-add "personal folder\.ssh\001\id_rsa"
  ssh-add "personal folder\clu\.ssh\2021-1025\id_rsa"
fi
```
